src/machine.cpp

```diff
@@ -444,28 +444,44 @@ machine::machine(const machine_config &c, const machine_runtime_config &r) :
     // Register pma board shadow device
     register_pma_entry(make_shadow_pmas_pma_entry(PMA_SHADOW_PMAS_START, PMA_SHADOW_PMAS_LENGTH));
 
-    // TODO(edubart): user should be able to configure these devices
-    if (m_c.htif.console_getchar) {
-        // Register VirtIO console device
-        auto vdev_console = std::make_unique<virtio_console>(m_vdevs.size());
-        register_pma_entry(
-            make_virtio_pma_entry(PMA_FIRST_VIRTIO_START + vdev_console->get_virtio_index() * PMA_VIRTIO_LENGTH,
-                PMA_VIRTIO_LENGTH, "VirtIO console device", &virtio_driver, vdev_console.get()));
-        m_vdevs.push_back(std::move(vdev_console));
-
-        // Register VirtIO Plan 9 filesystem device
-        auto vdev_p9fs = std::make_unique<virtio_p9fs_device>(m_vdevs.size(), "vfs0", "/tmp");
-        register_pma_entry(
-            make_virtio_pma_entry(PMA_FIRST_VIRTIO_START + vdev_p9fs->get_virtio_index() * PMA_VIRTIO_LENGTH,
-                PMA_VIRTIO_LENGTH, "VirtIO p9fs device", &virtio_driver, vdev_p9fs.get()));
-        m_vdevs.push_back(std::move(vdev_p9fs));
-
-        // Register VirtIO network device
-        auto vdev_net = std::make_unique<virtio_net>(m_vdevs.size(), std::make_unique<virtio_net_carrier_slirp>());
-        register_pma_entry(
-            make_virtio_pma_entry(PMA_FIRST_VIRTIO_START + vdev_net->get_virtio_index() * PMA_VIRTIO_LENGTH,
-                PMA_VIRTIO_LENGTH, "VirtIO network device", &virtio_driver, vdev_net.get()));
-        m_vdevs.push_back(std::move(vdev_net));
+    // Initialize VirtIO devices
+    if (!m_c.virtio.empty()) {
+        // VirtIO devices are disallowed when interactive mode is disabled
+        if (!m_c.htif.console_getchar) {
```

Comment:
I still don't think this makes much sense. There should be no connection between htif.console_getchar and virtio. They have the same "status". Did you fix this in some future commit already?

Reply:
You commented in an old commit, this was changed to check `iunrep` instead, although I forgot to change the comment.